src/composer/src/build.rs

```diff
@@ -371,14 +377,46 @@ impl BuildState for DefaultBuilder {
     fn comp_obj_path(&self, c: &ComponentId, s: &SystemState) -> Result<String, String> {
         self.comp_file_path(&c, &self.comp_obj_file(&c, &s), &s)
     }
+    
+    fn comp_const_h(
+        &self, id: &ComponentId, s: &SystemState, header_file: &mut Option<String>
+    ) -> Result<(), String> {
+        let c = component(&s, id);
+        let header_file_path = self.comp_file_path(id, &"component_constants.h".to_string(), &s)?;
+
+        if std::path::Path::new(&header_file_path).exists() {
+            //check the existing header file for constructor_build()
+            *header_file = Some(header_file_path);
```

Comment:
This is not the way to return multiple values in Rust.

Use the actual return value. You'll change `-> Result<(), String>` to `-> Result<String, String>` with the Ok branch returning the header file name as the `String`.